plugins/hfp_ag_bluez5.c

```diff
@@ -95,7 +109,10 @@ static DBusMessage *profile_new_connection(DBusConnection *conn,
 		goto invalid;
 
 	dbus_message_iter_get_basic(&entry, &fd);
+
 	dbus_message_iter_next(&entry);
+	if (dbus_message_iter_get_arg_type(&entry) != DBUS_TYPE_ARRAY)
```

Comment:
This isn't org.ofono.HandsfreeAudioAgent.NewConnection. It is the implementation of org.bluez.Profile1.NewConnection which takes an array as third paramter. See https://git.kernel.org/cgit/bluetooth/bluez.git/tree/doc/profile-api.txt#n119